packages/material-ui-lab/src/TreeNode/TreeNode.js

```diff
@@ -0,0 +1,322 @@
+import React from 'react';
+import clsx from 'clsx';
+import PropTypes from 'prop-types';
+import Typography from '@material-ui/core/Typography';
+import Collapse from '@material-ui/core/Collapse';
+import { withStyles } from '@material-ui/core/styles';
+import { useForkRef } from '@material-ui/core/utils';
+import TreeViewContext from '../TreeView/TreeViewContext';
+
+export const styles = {
+  /* Styles applied to the root element. */
+  root: {
+    listStyle: 'none',
+    margin: 0,
+    padding: 0,
+    outline: 0,
+    '&:focus > $content': {
+      outline: 'auto 1px',
+    },
+  },
+  /* Styles applied to the `role="group"` element. */
+  group: {
+    margin: 0,
+    padding: 0,
+    marginLeft: 26,
+  },
+  /* Styles applied to the tree node content. */
+  content: {
+    width: '100%',
+    display: 'flex',
+    alignItems: 'center',
+  },
+  /* Styles applied to the tree node icon and collapse/expand icon. */
+  iconContainer: {
+    marginRight: 2,
+    width: 24,
+    minWidth: 24,
+    display: 'flex',
+    justifyContent: 'center',
+  },
+  /* Styles applied to the label element. */
+  label: {},
+};
+
+const isPrintableCharacter = str => {
+  return str && str.length === 1 && str.match(/\S/);
+};
+
+const TreeNode = React.forwardRef(function TreeNode(props, ref) {
```

Comment:
Sorry about the late bike-shedding but what do you think about `TreeItem` following the aria role?

Reply:
Haha, I’m happy to change it to TreeItem 😜

Reply:
Love the idea :+1:.